subprojects/core-api/src/main/java/org/gradle/api/tasks/TaskOutputs.java

```diff
@@ -107,7 +107,10 @@
      * still have an empty set of output files.
      *
      * @return true if this task has declared any outputs, otherwise false.
+     *
+     * @deprecated Declare individual task properties to access output files.
```

Comment:
I guess the reason for deprecation here should be different.
I think that knowing the task type alone should answer the question if the task declared outputs or not. So there should be no need to call this method at all, right?

Reply:
What message would you prefer?

Reply:
I couldn't come up with a better message :(